Yoast/wordpress-seo
# Add IDs to the metabox and sidebar collapsibles

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* I've updated `react` and `react-dom` to `^16.4.1` to be able to use the context API . This is [the same version as Gutenberg is using](https://github.com/WordPress/gutenberg/blob/master/packages/element/package.json). We were already using this version in Gutenberg. Without this upgrade, there were errors in the classic editor. The React Tap Event Plugin, which we use in the configuration wizard, is [not compatible with this react version](https://github.com/zilverline/react-tap-event-plugin#deprecated). With approval of @atimmer I've removed that plugin. This results in a console warning on development, but not on production which has to do with the MaterialUI-based implementation of the step indicator:
<img width="867" alt="schermafbeelding 2018-10-10 om 15 11 39" src="https://user-images.githubusercontent.com/17744553/46738570-d7124d00-cc9e-11e8-89f9-6ca2662b7fa3.png">

Because we will move away from the step indicator/MaterialUI in the next version of the configuration wizard, we decided to leave this console warning for now.

* I've updated `enzyme-adapter-react-16` to `^1.5.0` to fix failing tests due to the react upgrade.
* I've replaced all occurrences of `this.props.location` by the context from the context API.

## Test instructions


This PR can be tested by following these steps:
- Link https://github.com/Yoast/yoast-components/pull/754 to this branch.
- Verify that all IDs from the following screenshots are there: 
<img width="1465" alt="46733497-d07cd900-cc90-11e8-9c3e-994060ba6bd9" src="https://user-images.githubusercontent.com/17744553/46750544-7bed5400-ccb8-11e8-8834-59fb30a38e3e.png">

- Verify that the upsells still have the correct links, because I've replaced the occurrence of `this.props.location` that were used there by the context from the context API. The new morphology upsells (sidebar and metabox), and these upsells:
<img width="442" alt="schermafbeelding_2018-10-10_om_14_20_32" src="https://user-images.githubusercontent.com/17744553/46735871-ca3e2b00-cc97-11e8-855b-d8b3cab92c48.png">

- There were some tricky merge conflicts. Make sure no UI elements are gone (for example question mark help links) or have reappeared (for example the language notice in the readability analysis)

* Test on posts, pages and taxonomies.
* Test in Gutenburg, classic and the classic editor plugin
- Because I've updated react, go to all screens that use react and check that nothing breaks: dashboard widget, configuration wizard, etc.
- **It is important to test Free as well as Premium** (see https://github.com/Yoast/wordpress-seo-premium/pull/2046)

## Quality assurance


Fixes #11177
